### DISCUSSION TITLE
Opacity/transparency not working at all; only makes background "whiter"

### DISCUSSION THREAD

**Discussion by tem** [ghostty-org/ghostty#3744] [Q&A] [answered]
**Note: Issue is fixed, see answer.**

~~**(Note: The issue is still persisting even after [this fix](https://github.com/ghostty-org/ghostty/commit/c8950d376ab3be13fbd9a19317163075f4feddbc) which closed issue #3324.)**~~

ghostty compiled from source via Zig x86_64 0.13.0. I'm on Linux (Ubuntu 22.04), KDE Plasma 5.24.7.

ghostty information (updated):

```
Ghostty 1.0.1-main+6508fec9 (previously 1.0.1-main+6cbd69da)

Version
  - version: 1.0.1-main+6508fec9 (previously 1.0.1-main+6cbd69da)
  - channel: tip
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.6.9
    runtime    : 4.6.9
  - libadwaita : enabled
    build      : 1.1.7
    runtime    : 1.1.7
  - libX11     : enabled
```

Before adding `background-opacity = 0.5`:

![ghostty-before](https://github.com/user-attachments/assets/ef632164-d1be-41c3-97c8-9a59d813491c)

After adding `background-opacity = 0.5`:

![ghostty-after](https://github.com/user-attachments/assets/b5e0f8ea-0951-42c2-9d8e-fadfa5997723)

Has anyone else faced this issue?

**Comment by rhodes-b** [comment]
can you show your the rest of your config. Also your GTK and adwaita versions are pretty old which may be the limiting factor here

**Reply by tem** [reply]
```
background-opacity = 0.5
```

is my entire config.

I'll look into upgrading my GTK and adwaita.

**Reply by tem** [reply]
Seems I have the latest versions of `libgtk-4-dev` and `libadwaita-1-dev` available on Ubuntu 22.04

**Comment by yspreen** [comment]
probably #3324

**Reply by rhodes-b** [reply]
Ah yeah I didnt notice the light theme

**Reply by tem** [reply]
The issue is still persisting even after [this fix](https://github.com/ghostty-org/ghostty/commit/c8950d376ab3be13fbd9a19317163075f4feddbc) which closed that issue (#3324).

**Comment by tem** [comment, marked as answer]
UPDATE!

I changed the config file and did a hot-reload with `Ctrl + Shift + ,`, and the aforementioned behavior persisted. HOWEVER, when I closed out of the application completely and re-opened it, not only did the opacity/transparency start working, but now hot-reloading also works (i.e., hot-reloading changes the opacity/transparency properly). Strange.

Version of ghostty used: `Ghostty 1.0.2-main+7eb35d72`
